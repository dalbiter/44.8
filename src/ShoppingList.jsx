import React, { useState } from 'react'
import Item from './Item';
import NewItemForm from './NewItemForm';
import {v4 as uuidv4} from 'uuid'

const ShoppingList = () => {
    const INITIAL_STATE = [
        {id: uuidv4(), name: "peanut butter", qty: 2},
        {id: uuidv4(), name: "whole milk", qty: 1}
    ]
    const [items, setItems] = useState(INITIAL_STATE)
    const addItem = (newItem) => {
        setItems(items => [...items, { ...newItem, id: uuidv4()}])
    }
    return (
        <div>
            <h3>Shopping List</h3>
            <NewItemForm addItem={addItem} />
            <div>
                {items.map(({ id, name, qty }) => <Item id={id} name={name} qty={qty} key={id} />)}
            </div>
        </div>
    )
};

export default ShoppingList;