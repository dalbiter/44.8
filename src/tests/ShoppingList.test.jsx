import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import ShoppingList from '../ShoppingList';
import { expect, it } from 'vitest';

it("renders without crashing", () => {
    render(<ShoppingList />);
});

it("matches snaphsot", () => {
    const { asFragment } = render(<ShoppingList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add new item", () => {
    const { queryByText, getByLabelText } = render(<ShoppingList />)
    const input = screen.getByLabelText('Product');
    const btn = queryByText("Add to List");
    expect(queryByText('Product Name: Chocolate Milk')).not.toBeInTheDocument();
    fireEvent.change(input, { target: { value: 'Chocolate Milk'} });
    fireEvent.submit(btn);
    expect(queryByText('Product Name: Chocolate Milk')).toBeInTheDocument();
});