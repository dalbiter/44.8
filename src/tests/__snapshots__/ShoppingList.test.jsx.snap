// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snaphsot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Shopping List
    </h3>
    <form>
      <label
        for="name"
      >
        Product
      </label>
      <input
        id="name"
        name="name"
        placeholder="Item Name"
        type="text"
        value=""
      />
      <label
        for="qty"
      >
        Quantity:  
      </label>
      <input
        max="10"
        min="1"
        name="qty"
        type="range"
        value="qty"
      />
      <button>
        Add to List
      </button>
    </form>
    <div>
      <ul>
        <li>
          Product Name: peanut butter
        </li>
        <li>
          Quantity: 2
        </li>
      </ul>
      <ul>
        <li>
          Product Name: whole milk
        </li>
        <li>
          Quantity: 1
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
